src/storage/table/row_group_collection.cpp

```diff
@@ -574,7 +575,9 @@ void RowGroupCollection::RemoveFromIndexes(TableIndexList &indexes, Vector &row_
 		result.Slice(sel, sel_count);
 
 		indexes.Scan([&](Index &index) {
-			index.Delete(result, row_identifiers);
+			if (index.IsBound()) {
+				index.Cast<BoundIndex>().Delete(result, row_identifiers);
```

Comment:
Should this not throw an exception if we have any unbound indexes? Otherwise the data in the index might not be kept up to date correctly.